src/main/java/io/github/moulberry/notenoughupdates/profileviewer/BasicPage.java

```diff
@@ -822,55 +824,88 @@ private void renderWeight(
 		SenitherWeight senitherWeight = new SenitherWeight(skyblockInfo);
 		LilyWeight lilyWeight = new LilyWeight(skyblockInfo, profileInfo);
 
-		Utils.drawStringCentered(
-			EnumChatFormatting.GREEN +
-				"Senither Weight: " +
-				EnumChatFormatting.GOLD +
-				GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight.getTotalWeight().getRaw())),
-			fr,
-			guiLeft + 63,
-			guiTop + 18,
-			true,
-			0
-		);
+		long weight = -2L;
+		if (NotEnoughUpdates.INSTANCE.config.profileViewer.useSoopyNetworth) {
+			weight = profile.getSoopyWeightLeaderboardPosition();
```

Comment:
If i turn off soopy networth in the settings it just gets stuck on "Loading...". Maybe also: since we already know how to calculate the senither weight accurately we might as well show the weight all the time and only display "Loading..." on hover in the leaderboard place